src/D2L.Bmx/Okta/OktaSessionStorage.cs

```diff
@@ -0,0 +1,67 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using D2L.Bmx.Okta.Models;
+
+namespace D2L.Bmx.Okta;
+
+[JsonSerializable( typeof( OktaSessionCache[] ) )]
+internal partial class OktaSessionContext : JsonSerializerContext {
+}
+
+internal class OktaSessionStorage {
+
+	internal static void SaveSessions( List<OktaSessionCache> sessions ) {
+
+		string jsonString = JsonSerializer.Serialize( sessions.ToArray(), OktaSessionContext.Default.OktaSessionCacheArray );
+		File.WriteAllText( SessionsFileName(), jsonString );
+	}
+
+	internal static List<OktaSessionCache> Sessions() {
+
+		var bmxDirectory = BmxDir();
+		var sessionsFileName = SessionsFileName();
+
+		if( !Directory.Exists( bmxDirectory ) ) {
+			Directory.CreateDirectory( bmxDirectory );
+		}
+
+		if( !File.Exists( sessionsFileName ) ) {
+			File.WriteAllText( sessionsFileName, "[]" );
+		}
```

Comment:
We talked about this in pair programming. Wanting to create the sessions file in case it wasn't created. Did the directory as well to be safe?

Reply:
Looks necessary for if the directory doesn't exist, WriteAllText throws an exception otherwise.